lib/ex_doc/doc_ast.ex

```diff
@@ -4,6 +4,12 @@ defmodule ExDoc.DocAST do
   alias ExDoc.Markdown
   alias ExDoc.Formatter.HTML.Templates
 
+  @opaque t() :: [ast]
+
+  @typep ast() ::
+           {tag :: atom(), attrs :: [{atom(), binary()}], [ast()], meta :: map()}
+           | binary()
+
```

Comment:
Should we move this module inside `ExDoc.Language`?

Reply:
I wanted to keep the namespaces flatter, doc ast to me is orthogonal to language contract, but at the same time we have this unified doc ast exactly because of language contract. I stil wonder if this DocAST would ever become public, I think it would be nice to refer to `DocAST.t` from `ExDoc.Markdown` contract - you need to know the shape of the thing your callbacks are supposed to return. Of course we could export a separate ExDoc.Markdown.ast type and keep DocAST internal.

So yeah, I think it's fine to make this `ExDoc.Language.DocAST`, we can always change our mind later.